httpcore/_async/http_proxy.py

```diff
@@ -228,6 +238,7 @@ def __init__(
         http2: bool = False,
         network_backend: Optional[AsyncNetworkBackend] = None,
         socket_options: Optional[Iterable[SOCKET_OPTION]] = None,
+        is_tls: bool = True,
```

Comment:
Do we really need this? We can simply check if `remote_origin` is https or not

Reply:
yes, makes sense, thanks for review